x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/graph_investigation.tsx

```diff
@@ -309,6 +208,21 @@ export const GraphInvestigation = memo<GraphInvestigationProps>(
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [data?.nodes]);
 
+    const searchFilterCounter = useMemo(() => {
+      const filtersCount = searchFilters
+        .filter((filter) => !filter.meta.disabled)
+        .reduce((sum, filter) => {
+          if (isCombinedFilter(filter)) {
+            return sum + filter.meta.params.length;
+          }
+
+          return sum + 1;
+        }, 0);
+
+      const queryCounter = kquery.query.trim().length > 0 ? 1 : 0;
```

Comment:
I guess this is the logic to count any query as one filter, which I found confusing. I kind of get the idea, though counting even complex queries as 1 is counter intuitive. Don't know if we have an unitility that can give us the amount of filters in the query and maybe it's overengineering, but one option to consider